File: InternHub/InternHub.Model/InternshipApplication.cs
using InternHub.Model.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InternHub.Model
{
    public class InternshipApplication : BaseModel, IInternshipApplication
    {
     
        public Guid StateId { get; set; }
        public State State { get; set; }
        public string StudentId { get; set; }
        public Student Student { get; set; }
        public Guid InternshipId { get; set; }
        public Internship Internship { get; set; }
        public string Message { get; set; } 

    }
}

File: internhub-frontend/src/models/InternshipApplication.js
import { Student, State, Internship } from "./index";

export class InternshipApplication {
  constructor({
    id = "",
    dateCreated = null,
    dateUpdated = null,
    message = "",
    student = null,
    state = null,
    internship = null,
  }) {
    this.id = id;
    this.dateCreated = dateCreated;
    this.dateUpdated = dateUpdated;
    this.message = message;
    this.student = student;
    this.state = state;
    this.internship = internship;
  }

  static fromJson(json) {
    return new InternshipApplication({
      id: json["Id"] || "",
      dateCreated: json["DateCreated"] || "",
      dateUpdated: json["DateUpdated"] || "",
      message: json["Message"] || "",
      student: json["Student"] ? Student.fromJson(json["Student"]) : null,
      state: json["State"] ? State.fromJson(json["State"]) : null,
      internship: json["Internship"]
        ? Internship.fromJson(json["Internship"])
        : json["Internship"],
    });
  }
}
